daal4py/sklearn/neighbors/knn.py

```diff
@@ -32,6 +32,11 @@
 
 class KNeighborsMixin(BaseKNeighborsMixin):
     def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
+        n_features = getattr(self, 'n_features_in_', None)
+        shape = getattr(X, 'shape', None)
+        if n_features and shape and shape[1] != n_features:
```

Comment:
len(shape) might be 1 if 1d array is passed and error will be raised while trying to get shape[1]

Reply:
Additional check was added